larsim/LegacyLArG4/OpDetSensitiveDetector.cxx

```diff
@@ -36,28 +39,42 @@ namespace larg4{
 
   //--------------------------------------------------------
 
-  G4bool OpDetSensitiveDetector::ProcessHits(G4Step * aStep, G4TouchableHistory *)
+  void OpDetSensitiveDetector::AddLitePhoton(G4Step const* aStep, int OpDet)
   {
-    sim::OnePhoton ThePhoton;
+    
+    double const time  = aStep->GetTrack()->GetGlobalTime();
+
+    // the guideline: if it's VUV (~128 nm) is direct, otherwise it is reflected
+    double const energy = aStep->GetTrack()->GetVertexKineticEnergy() / CLHEP::eV;
+    bool const reflected = Wavelength(energy) > 200.0; // nm
+    
+    // Add this photon to the detected photons table
+    fThePhotonTable->AddLitePhoton(OpDet, static_cast<int>(time), 1, reflected);
+    
+  } // OpDetSensitiveDetector::AddLitePhoton()
 
+  
+  //--------------------------------------------------------
+
+  void OpDetSensitiveDetector::AddPhoton(G4Step const* aStep, int OpDet)
+  {
+    sim::OnePhoton ThePhoton;
 
     // Get photon data to store in the hit
 
     ThePhoton.SetInSD      = true;
 
-    auto const& startPos = aStep->GetTrack()->GetVertexPosition();
+    auto const& track = *(aStep->GetTrack());
+    auto const& startPos = track.GetVertexPosition();
     ThePhoton.InitialPosition = { startPos.x(), startPos.y(), startPos.z() };
 
-    //ThePhoton.Time                = aStep->GetTrack()->GetGlobalTime() - fGlobalTimeOffset;
-    ThePhoton.Time                = aStep->GetTrack()->GetGlobalTime();
-
+    //ThePhoton.Time                = track.GetGlobalTime() - fGlobalTimeOffset;
+    ThePhoton.Time                = track.GetGlobalTime();
 
-    ThePhoton.Energy              =  aStep->GetTrack()->GetVertexKineticEnergy();
+    ThePhoton.Energy              =  track.GetVertexKineticEnergy();
```

Comment:
Can `ThePhoton` be initialized at its point of declaration?

Reply:
It is, with an aggregate initialization.
I try to avoid that (until C++20) because a change in the definition may mess up the initialization (in this case, if for any reason `Energy` and `Time` are swapped).
I suggest we leave it this way, but I let you have the last word on the topic.

Reply:
It's a minor inefficiency but not worth holding up the PR.